observability/src/main/java/io/grpc/observability/interceptors/ConfigFilterHelper.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2022 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.observability.interceptors;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.UnmodifiableListIterator;
+import io.grpc.MethodDescriptor;
+import io.grpc.observability.ObservabilityConfig;
+import io.grpc.observability.ObservabilityConfig.LogFilter;
+import io.grpc.observabilitylog.v1.GrpcLogRecord.EventType;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class ConfigFilterHelper {
```

Comment:
- this is public because it is used from `Observability`. Should this be then moved to that package?
- otherwise annotate as Internal and make it final
- Javadoc for what this class does

Reply:
Added Javadoc and marked the class internal. Leaving it in interceptors package as per our discussion offline.

Reply:
Also cannot be final because of the need to mock? I wonder if `spy` can be used instead.